File: 02_Object_Oriented_Programming/02_Class_Relationships/02_Association/javascript/car.js
/**
 * Association — Car class in JavaScript.
 */

class Car {
  #model;

  constructor(model) {
    this.#model = model;
  }

  get model() {
    return this.#model;
  }

  drive() {
    console.log(`Driving a ${this.#model}`);
  }

  toString() {
    return `Car(${this.#model})`;
  }
}

module.exports = Car;

File: 02_Object_Oriented_Programming/02_Class_Relationships/02_Association/java/Car.java
/**
 * Association — Car class.
 *
 * Key concept: A Person HAS-A Car (uses it, but doesn't own its lifecycle).
 * Both objects can exist independently.
 */
public class Car {

    private String model;

    public Car(String model) {
        this.model = model;
    }

    public String getModel() {
        return model;
    }

    public void drive() {
        System.out.println("Driving a " + model);
    }

    @Override
    public String toString() {
        return "Car(" + model + ")";
    }
}
